test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,291 @@
+	// avoid error `no kind is registered for the type v1.ListOptions`
+	_ "k8s.io/kubernetes/pkg/api/install"
+
+	// to start our server locally
+	"github.com/kubernetes-incubator/service-catalog/cmd/apiserver/app/server"
+	// our versioned types
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	// our versioned client
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+)
+
+var (
+	kubeconfig = flag.String("kubeconfig", ".kube/config", "absolute path to the kubeconfig file")
+)
+
+func TestRunServer(t *testing.T) {
+
+}
+
+func testDirectCodec(t *testing.T) {
+	flag.Parse()
+
+	// run etcd?
+	// run server?
+
+	// config to talk to server. localhost 6443 with certificate /var/run/kubernetes ?
+
+	// write our own config ?
+	//client := clientset.NewForConfigOrDie(&restclient.Config{Host: "", ContentConfig: restclient.ContentConfig{GroupVersion: &registered.GroupOrDie(v1.GroupName).GroupVersion}})
+
+	// relies on set up from the infrastructure
+	absKConfig, err := filepath.Abs(*kubeconfig)
+	if nil != err {
+		glog.Fatal("could not load kubeconfig")
+	}
+	// uses the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", absKConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+	// creates the clientset
+	clientset, err := servicecatalog.NewForConfig(config)
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	//for {
+	// we currently fail at the 'List()' part.
+	brokers, err := clientset.Servicecatalog().Brokers().List(v1.ListOptions{})
+	if err != nil {
+		glog.Flush()
+		glog.Fatal(err)
+	}
+	fmt.Printf("There are %d brokers in the cluster\n", len(brokers.Items))
+	time.Sleep(10 * time.Second)
+	//}
+}
+
+func TestRunSecureServer(t *testing.T) {
+	securePort := 8888
+	serverIP := fmt.Sprintf("https://localhost:%d", securePort)
+	stopCh := make(chan struct{})
+	serverFailed := make(chan struct{})
+	defer close(stopCh)
+
+	secureServingOptions := genericserveroptions.NewSecureServingOptions()
+	go func() {
+		options := &server.ServiceCatalogServerOptions{
+			GenericServerRunOptions: genericserveroptions.NewServerRunOptions(),
+			SecureServingOptions:    secureServingOptions,
+			EtcdOptions:             genericserveroptions.NewEtcdOptions(),
+			AuthenticationOptions:   genericserveroptions.NewDelegatingAuthenticationOptions(),
+			AuthorizationOptions:    genericserveroptions.NewDelegatingAuthorizationOptions(),
+		}
+		options.SecureServingOptions.ServingOptions.BindPort = securePort
+		options.EtcdOptions.StorageConfig.ServerList = []string{"http://localhost:2379"}
+		if err := options.RunServer(stopCh); err != nil {
+			close(serverFailed)
+			t.Fatalf("Error in bringing up the server: %v", err)
+		}
+	}()
+
+	if err := waitForApiserverUp(serverIP, serverFailed); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	testSwaggerSpec(t, serverIP)
+	testAPIGroupList(t, serverIP)
+	testAPIGroup(t, serverIP)
+	testAPIResourceList(t, serverIP)
```

Comment:
this group of four was copied from upstream. they used the testify library. Not sure if we want that dependency.